pkg/store/storepb/custom.go

```diff
@@ -456,3 +461,114 @@ func CompareLabels(a, b []Label) int {
 func LabelsToPromLabelsUnsafe(lset []Label) labels.Labels {
 	return labelpb.ZLabelsToPromLabels(lset)
 }
+
+// Type alias because protoc-go-inject-field does not support
+// managing imports.
+type syncPool = sync.Pool
+
+// The following were copied/pasted from gogoprotobuf generated code with changes
+// to make it work with sync.Pool / []byte slice.
+func (m *SeriesResponse) Marshal() (dAtA []byte, err error) {
+	size := m.Size()
+
+	var respBuf []byte
+	if m.respBuf == nil {
+		respBuf = make([]byte, size)
+	} else {
+		if cap(**m.respBuf) < size {
+			if m.respPool != nil {
+				m.respPool.Put(*m.respBuf)
+			}
+			buf := make([]byte, size)
+			*m.respBuf = &buf
+		}
+		respBuf = **m.respBuf
+	}
+
+	marshalBuf := respBuf[:size]
+	n, err := m.MarshalToSizedBuffer(marshalBuf)
+	if err != nil {
+		return nil, err
+	}
+	return marshalBuf[len(marshalBuf)-n:], nil
+}
+
+func (m *SeriesResponse) MarshalTo(dAtA []byte) (int, error) {
+	size := m.Size()
+	return m.MarshalToSizedBuffer(dAtA[:size])
+}
+
+type marshaler interface {
+	MarshalTo([]byte) (int, error)
+}
+
+func (m *SeriesResponse) MarshalToSizedBuffer(data []byte) (int, error) {
+	i := len(data)
+
+	if m.Result != nil {
+		size := m.Result.Size()
+		i -= size
+
+		if _, err := m.Result.(marshaler).MarshalTo(data[i:]); err != nil {
+			return 0, err
+		}
+	}
+	return len(data) - i, nil
+}
+
+func (m *SeriesResponse_Series) MarshalTo(data []byte) (int, error) {
+	size := m.Size()
+	return m.MarshalToSizedBuffer(data[:size])
+}
+
+func (m *SeriesResponse_Series) MarshalToSizedBuffer(data []byte) (int, error) {
+	i := len(data)
+	if m.Series != nil {
+		{
+			size, err := m.Series.MarshalToSizedBuffer(data[:i])
+			if err != nil {
+				return 0, err
+			}
+			i -= size
+			i = encodeVarintRpc(data, i, uint64(size))
+		}
+		i--
+		data[i] = 0xa
+	}
+	return len(data) - i, nil
+}
+func (m *SeriesResponse_Warning) MarshalTo(data []byte) (int, error) {
+	size := m.Size()
+	return m.MarshalToSizedBuffer(data[:size])
+}
+
+func (m *SeriesResponse_Warning) MarshalToSizedBuffer(data []byte) (int, error) {
+	i := len(data)
+	i -= len(m.Warning)
+	copy(data[i:], m.Warning)
+	i = encodeVarintRpc(data, i, uint64(len(m.Warning)))
+	i--
+	data[i] = 0x12
+	return len(data) - i, nil
+}
+func (m *SeriesResponse_Hints) MarshalTo(data []byte) (int, error) {
+	size := m.Size()
+	return m.MarshalToSizedBuffer(data[:size])
+}
+
+func (m *SeriesResponse_Hints) MarshalToSizedBuffer(data []byte) (int, error) {
```

Comment:
Those are changed or just copied?

Reply:
These are copied and minimally cleaned up as per @kakkoyun's suggestion.